app/src/main/java/org/sil/storyproducer/controller/SelectTemplatesFolderController.kt

```diff
@@ -18,32 +20,61 @@ class SelectTemplatesFolderController(
         view.startActivityForResult(intent, request)
     }
 
+    private fun showMigrateIntenalStoriesDialog(request: Int, uri: Uri) {
+        val dialog = AlertDialog.Builder(context)
+            .setTitle(context.getString(R.string.migrate_stories_title))
+            .setMessage(context.getString(R.string.migrate_stories_message))
+            .setPositiveButton(context.getString(R.string.yes)) { _, _ ->
+                setupWorkspace(request, uri, true)
+            }
+            .setNegativeButton(context.getString(R.string.no)) { _, _ ->
+                setupWorkspace(request, uri, false)
+            }
+            .setOnCancelListener() {
+                setupWorkspace(request, uri, false)
+            }
+            .create()
+
+        dialog.show()
+    }
+
     fun onFolderSelected(request: Int, result: Int, data: Intent?) {
         data?.data?.also { uri ->
             if (result == Activity.RESULT_OK) {
-                setupWorkspace(request, uri)
+                if (!workspace.setupWorkspacePath(context, uri, false))
+                    return
+                if (//workspace.workdocfile.uri.scheme == "file" && // TODO: Should we only migrate from internal folders?
+                        workspace.previousWorkDocFile.exists() &&
+                        workspace.previousWorkDocFile.uri != workspace.workdocfile.uri &&
+                        workspace.oldStoryDirectories(workspace.storyDirectories(), true).isNotEmpty()) {
+                    showMigrateIntenalStoriesDialog(request, uri)
+                } else {
+                    setupWorkspace(request, uri, false)
+                }
             }
         }
     }
 
-    internal fun setupWorkspace(request: Int, uri: Uri) {
+    internal fun setupWorkspace(request: Int, uri: Uri, migrate: Boolean = false) {
         view.takePersistableUriPermission(uri)
 
-        workspace.setupWorkspacePath(context, uri)
+        if (!workspace.setupWorkspacePath(context, uri))
+            return
 
         if (shouldAddDemoToWorkspace(request)) {
             workspace.addDemoToWorkspace(context)
         }
 
-        updateStories()  // always refresh list of stories
+        updateStories(migrate)  // always refresh list of stories
     }
 
     fun shouldAddDemoToWorkspace(request: Int): Boolean {
         if (request == SELECT_TEMPLATES_FOLDER_AND_ADD_DEMO)
             return true
 
         // always add demo when no installed stories or stories to unzip
-        if (workspace.storyFilesToScanOrUnzip().isEmpty())
+        if (Workspace.workdocfile.uri.scheme == "file" &&   // TODO: Should we only create a demo in internal folders?
```

Comment:
TODO: Should we only create a demo in automatic folders?

Reply:
Added a comment to 'Automatic workspace folder' document to cover this